lib/Core/BuildEngine.cpp

```diff
@@ -479,6 +483,7 @@ class BuildEngineImpl : public BuildDBDelegate {
       if (trace)
         trace->ruleNeedsToRunBecauseInvalidValue(ruleInfo.rule.get());
```

Comment:
Would be cool if the trace would also use the new enum

Reply:
That's a good idea. I'll file a follow-up bug for the refactor